src/util/decode_encode_muxed_account.js

```diff
@@ -86,6 +70,20 @@ export function encodeMuxedAccount(address, id) {
   );
 }
 
+/**
+ * Extracts the underlying base (G...) address from an M-address.
+ * @param  {string} address  a muxed account address (M...)
+ * @return {string} a Stellar public key address (G...)
+ */
+export function extractBaseAddress(address) {
+  if (!StrKey.isValidMed25519PublicKey(address)) {
+    throw new TypeError('address should be a muxed account (M...)');
+  }
```

Comment:
❔ Is there a way to test if an address is a muxed address before passing it to this function? I'm wondering about the flow where someone has a value that might be an M or a G and they want to get the G. If they call this function on its own it might error in the G cases. How do they distinguish? Do they just check the first character? Or maybe this function should just return the G if the input is a G. I don't feel strongly what way we go, but curious what the story is for this use case.

Reply:
Haha, I realize the line I'm commenting on literally has the function for testing if an address is a muxed address.

Reply:
I still wonder if there is value in not erroring in the case a G is passed and just returning the G. It might make the API a little easier to use, albeit with no errors in a case that might be error worthy.

Reply:
That's an interesting question. I can't foresee when it would come up, but I don't have a strong opinion either way.

Callers can use the `StrKey` validators (`isValidEd25519PublicKey` and `isValidMed25519PublicKey`, just like is done here) before calling the function. I added a check in 9e1ee1f to just pass G-addresses through, for convenience.

Reply:
Hah, should have refreshed before replying :ninja:. But yeah, added the pass-through.